app/screens/channel_files/channel_files.tsx

```diff
@@ -98,7 +100,18 @@ function ChannelFiles({
         const t = Date.now();
         lastSearchRequest.current = t;
         const searchParams = getSearchParams(channel.id, searchTerm, ftr);
-        const {files} = await searchFiles(serverUrl, channel.teamId, searchParams);
+        let files: FileInfo[] | undefined = [];
+        if (channel.teamId) {
+            const searchedFiles = await searchFiles(serverUrl, channel.teamId, searchParams);
+            files = searchedFiles.files;
+        } else {
+            // teamId can be empty for GMs and DMs, as do not store teamId for GMs and DMs in the database
+            // so we need to fetch the current teamId from the systems table
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const teamId = await getCurrentTeamId(database);
+            const searchedFiles = await searchFiles(serverUrl, teamId, searchParams);
```

Comment:
Instead of doing repeating the api call twice we can just first get team id first and then do the api call

Reply:
nit, just for DRY:

```
            let teamId = channel.teamId;
            if (!teamId) {
                const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
                teamId = await getCurrentTeamId(database);
            }

            const searchedFiles = await searchFiles(serverUrl, teamId, searchParams);
            files = searchedFiles.files;
```

Reply:
Oh @M-ZubairAhmed, I think that's what you meant, right?  My apologies, then my comment maybe redundant.

Reply:
I changed the approach.